tests/testing_utils.py

```diff
@@ -412,7 +413,8 @@ def optimize_with_hotstart(self, tol, optOptions=None, x0=None):
         sol = self.optimize(storeHistory=True, optOptions=optOptions, setDV=x0)
         self.assert_solution_allclose(sol, tol)
         self.assertGreater(self.nf, 0)
-        self.assertGreater(self.ng, 0)
+        if self.optName != "ALPSO":
```

Comment:
Instead of adding these checks specifically for ALPSO, I would prefer if we defined a set of gradient-free optimizers after [this line](https://github.com/Laurentww/pyoptsparse/blob/master/tests/testing_utils.py#L129), e.g. `GRAD_FREE_OPTIMIZERS = {"NSGA2", "ALPSO"}`, then we can check with `if self.optName in GRAD_FREE_OPTIMIZERS:`

Reply:
That clarifies it nicely, thanks for suggesting. I added it in the last commit.